src/hotspot/share/c1/c1_Runtime1.cpp

```diff
@@ -505,6 +506,21 @@ JRT_ENTRY_NO_ASYNC(static address, exception_handler_for_pc_helper(JavaThread* t
   thread->set_is_method_handle_return(false);
 
   Handle exception(thread, ex);
+
+  // This function is called when we are about to throw an exception. Therefore,
+  // we have to poll the stack watermark barrier to make sure that not yet safe
+  // stack frames are made safe before returning into them.
+  if (thread->last_frame().is_runtime_frame()) {
+    // The Runtime1::handle_exception_from_callee_id handler is invoked after the
+    // frame has been unwinded. It instead builds its own stub frame, to call the
```

Comment:
Unwided -> unwound, maybe? Same below and probably other places as well.

Reply:
Good point.